From 31bd3a0996a85c0fced0c6ace3da1241b30dc397 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20Men=C5=A1=C3=ADk?= <pemensik@redhat.com>
Date: Thu, 30 Apr 2026 17:28:40 +0200
Subject: [PATCH] Use variable PROGRAM_SUFFIX in install target

--program-suffix is handled by configure itself. But some makefile rules
need to know it by a special value. Provide that to make multiple
versions installable together on a single system.
---
 Makefile.top            | 1 +
 bin/check/Makefile.am   | 6 +++---
 bin/confgen/Makefile.am | 6 +++---
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/Makefile.top b/Makefile.top
index e186d15..91c076b 100644
--- a/Makefile.top
+++ b/Makefile.top
@@ -14,6 +14,7 @@ AM_CPPFLAGS =					\
 AM_LDFLAGS =					\
 	$(STD_LDFLAGS)
 LDADD =
+PROGRAM_SUFFIX =
 
 if HOST_MACOS
 AM_LDFLAGS +=					\
diff --git a/bin/check/Makefile.am b/bin/check/Makefile.am
index 8f63c35..36f232c 100644
--- a/bin/check/Makefile.am
+++ b/bin/check/Makefile.am
@@ -27,8 +27,8 @@ LDADD +=			\
 bin_PROGRAMS = named-checkconf named-checkzone
 
 install-exec-hook:
-	ln -f $(DESTDIR)$(bindir)/named-checkzone \
-	      $(DESTDIR)$(bindir)/named-compilezone
+	ln -f $(DESTDIR)$(bindir)/named-checkzone$(PROGRAM_SUFFIX) \
+	      $(DESTDIR)$(bindir)/named-compilezone$(PROGRAM_SUFFIX)
 
 uninstall-hook:
-	-rm -f $(DESTDIR)$(bindir)/named-compilezone
+	-rm -f $(DESTDIR)$(bindir)/named-compilezone$(PROGRAM_SUFFIX)
diff --git a/bin/confgen/Makefile.am b/bin/confgen/Makefile.am
index c1dca43..fe86dd7 100644
--- a/bin/confgen/Makefile.am
+++ b/bin/confgen/Makefile.am
@@ -23,8 +23,8 @@ libconfgen_la_SOURCES =		\
 sbin_PROGRAMS = tsig-keygen rndc-confgen
 
 install-exec-hook:
-	ln -f $(DESTDIR)$(sbindir)/tsig-keygen \
-	      $(DESTDIR)$(sbindir)/ddns-confgen
+	ln -f $(DESTDIR)$(sbindir)/tsig-keygen$(PROGRAM_SUFFIX) \
+	      $(DESTDIR)$(sbindir)/ddns-confgen$(PROGRAM_SUFFIX)
 
 uninstall-hook:
-	-rm -f $(DESTDIR)$(sbindir)/ddns-confgen
+	-rm -f $(DESTDIR)$(sbindir)/ddns-confgen$(PROGRAM_SUFFIX)
-- 
2.54.0

